Test/FMI3/fmi3_import_model_structure_test.cpp

```diff
@@ -244,200 +237,305 @@ TEST_CASE("Valid ModelStructure parsing") {
 }
 
 TEST_CASE("Error check: ModelStructure; EventIndicator variability") {
     const char* xmldir = FMI3_TEST_XML_DIR "/model_structure/invalid/event_indicator_variability";
-
     fmi3_testutil_import_t* tfmu = fmi3_testutil_parse_xml_with_log(xmldir);
     REQUIRE(tfmu != nullptr);
-    REQUIRE(tfmu->fmu == nullptr);
+    fmi3_import_t* fmu = tfmu->fmu;
+    REQUIRE(fmu != nullptr);
 
-    const char* errMsg = "The variable 'event_ind' is an EventIndicator, but does not have variability='continuous'.";
-    REQUIRE(fmi3_testutil_log_contains(tfmu, errMsg));
+    REQUIRE(fmi3_testutil_log_contains(tfmu, "The variable 'event_ind' is an EventIndicator, but does not have variability='continuous'."));
+
+    fmi3_import_variable_list_t* varList = fmi3_import_get_event_indicators_list(fmu);
+    REQUIRE(fmi3_import_get_variable_list_size(varList) == 1);
+    fmi3_import_variable_t* var = fmi3_import_get_variable(varList, 0);
+    REQUIRE(var != nullptr);
+    REQUIRE(fmi3_import_get_variable_variability(var) == fmi3_variability_enu_discrete);
+
+    fmi3_import_free_variable_list(varList);
 
     fmi3_testutil_import_free(tfmu);
 }
 
 TEST_CASE("Error check: ModelStructure; EventIndicator base type") {
     const char* xmldir = FMI3_TEST_XML_DIR "/model_structure/invalid/event_indicator_base_type";
-
     fmi3_testutil_import_t* tfmu = fmi3_testutil_parse_xml_with_log(xmldir);
     REQUIRE(tfmu != nullptr);
-    REQUIRE(tfmu->fmu == nullptr);
-
-    const char* errMsg = "The variable 'event_ind' is an EventIndicator, but does not have the base type 'Float32' or 'Float64'.";
-    REQUIRE(fmi3_testutil_log_contains(tfmu, errMsg));
-
-    fmi3_testutil_import_free(tfmu);
-}
+    fmi3_import_t* fmu = tfmu->fmu;
+    REQUIRE(fmu != nullptr);
 
-TEST_CASE("Info check: ModelStructure; EventIndicator ignored for Co-Simulation") {
-    const char* xmldir = FMI3_TEST_XML_DIR "/model_structure/valid/event_indictor_CS";
+    REQUIRE(fmi3_testutil_log_contains(tfmu, "The variable 'event_ind' is an EventIndicator, but does not have the base type 'Float32' or 'Float64'."));
 
-    fmi3_testutil_import_t* tfmu = fmi3_testutil_parse_xml_with_log(xmldir);
-    REQUIRE(tfmu != nullptr);
-    REQUIRE(tfmu->fmu != nullptr); // Parsing successful
+    fmi3_import_variable_list_t* varList = fmi3_import_get_event_indicators_list(fmu);
+    REQUIRE(fmi3_import_get_variable_list_size(varList) == 1);
+    fmi3_import_variable_t* var = fmi3_import_get_variable(varList, 0);
+    REQUIRE(var != nullptr);
+    fmi3_import_int64_variable_t* i64Var = fmi3_import_get_variable_as_int64(var);
+    REQUIRE(var != nullptr);
 
-    const char* logMsg = "EventIndicator ignored since FMU kind is Co-Simulation or Scheduled Excecution.";
-    REQUIRE(fmi3_testutil_log_contains(tfmu, logMsg));
+    fmi3_import_free_variable_list(varList);
 
     fmi3_testutil_import_free(tfmu);
 }
 
-TEST_CASE("Error check: ModelStructure; ClockedState without clocks attribute") {
```

Comment:
This test was redundant; the check is already covered by attribute combinations checks in ModelVariables.